src/trait-bounds.md

```diff
@@ -19,6 +19,20 @@
 > _Lifetime_ :\
 > &nbsp;&nbsp; &nbsp;&nbsp; [LIFETIME_OR_LABEL]\
 > &nbsp;&nbsp; | `'static`
+>
+> _UseBound_ :\
+> &nbsp;&nbsp; `use` _UseBoundGenericArgs_
+>
+> _UseBoundGenericArgs_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; `<` `>` \
+> &nbsp;&nbsp; | `<` \
+> &nbsp;&nbsp; &nbsp;&nbsp; ( _UseBoundGenericArg_ `,`)<sup>\*</sup> \
+> &nbsp;&nbsp; &nbsp;&nbsp; _UseBoundGenericArg_ `,`<sup>?</sup> \
+> &nbsp;&nbsp; &nbsp;&nbsp; `>`
+>
+> _UseBoundGenericArg_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; [LIFETIME_OR_LABEL][] \
+> &nbsp;&nbsp; | [IDENTIFIER][]
```

Comment:
What's the story about `Self` here? It looks like the compiler accepts it, but it is then rejected?

Reply:
`Self` is allowed semantically when it's a *parameter*, but the only case where that is true is when it's in an RPITIT, which we don't allow `use<>` on yet.

Reply:
Syntactically `Self` is just an identifier, so it should be allowed. But that's the same as non-param lifetimes like `'static`, which are allowed syntactically but not semantically.

Reply:
The reference excludes keywords from IDENTIFIER, and `Self` is a keyword.

I guess similarly for `'static`, that would mean this accepts _Lifetime_, not LIFETIME_OR_LABEL.

Is there a timeline for supporting RPITIT?

Reply:
Lifetime_or_label is a subset of lifetime? Confusing

> Is there a timeline for supporting RPITIT?

Nope

Reply:
> The reference excludes keywords from IDENTIFIER, and Self is a keyword.

Well, so the only keyword that we allow here is `Self`, so let's make it `IDENTIFIER | Self | LIFETIME`.

Reply:
Thanks; updated;